examples/analytics/airflow-on-eks/main.tf

```diff
@@ -0,0 +1,542 @@
+
+  depends_on = [aws_secretsmanager_secret_version.postgres]
+}
+
+#---------------------------------------------------------------
+# Apache Airflow Webserver Secret
+#---------------------------------------------------------------
+resource "random_password" "airflow_webserver" {
+  length           = 16
+  special          = true
+  override_special = "![]{}"
+}
+#tfsec:ignore:aws-ssm-secret-use-customer-key
+resource "aws_secretsmanager_secret" "airflow_webserver" {
+  name                    = "airflow_webserver"
+  recovery_window_in_days = 0 # Set to zero for this example to force delete during Terraform destroy
+}
+
+resource "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id     = aws_secretsmanager_secret.airflow_webserver.id
+  secret_string = random_password.airflow_webserver.result
+}
+
+data "aws_secretsmanager_secret_version" "airflow_webserver" {
+  secret_id = aws_secretsmanager_secret.airflow_webserver.id
+
+  depends_on = [aws_secretsmanager_secret_version.airflow_webserver]
+}
+
+#---------------------------------------------------------------
+# Webserver Secret Key
+#---------------------------------------------------------------
+resource "kubectl_manifest" "airflow_webserver" {
+  sensitive_fields = [
+    "data.webserver-secret-key"
+  ]
+
+  yaml_body = <<YAML
+apiVersion: v1
+kind: Secret
+metadata:
+   name: ${local.airflow_webserver_secret_name}
+   namespace: ${module.airflow_irsa.namespace}
+type: Opaque
+data:
+  webserver-secret-key: ${base64encode(data.aws_secretsmanager_secret_version.airflow_webserver.secret_string)}
+YAML
+
+  depends_on = [module.eks_blueprints.eks_cluster_id]
+}
+
+#---------------------------------------------------------------
+# Managing DAG files with GitSync - EFS Storage Class
+#---------------------------------------------------------------
+resource "kubectl_manifest" "efs_sc" {
+  yaml_body = <<YAML
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: efs-sc
+provisioner: efs.csi.aws.com
+parameters:
+  provisioningMode: efs-ap
+  fileSystemId: ${aws_efs_file_system.efs.id}
+  directoryPerms: "700"
+  gidRangeStart: "1000"
+  gidRangeEnd: "2000"
+YAML
+
+  depends_on = [module.eks_blueprints.eks_cluster_id]
+}
+#---------------------------------------------------------------
+# EFS Filesystem for Airflow DAGs
+#---------------------------------------------------------------
+resource "aws_efs_file_system" "efs" {
+  creation_token = "efs"
+  encrypted      = true
+
+  tags = local.tags
+}
+
+resource "aws_efs_mount_target" "efs_mt" {
+  count = length(module.vpc.private_subnets)
+
+  file_system_id  = aws_efs_file_system.efs.id
+  subnet_id       = module.vpc.private_subnets[count.index]
+  security_groups = [aws_security_group.efs.id]
+}
+
+resource "aws_security_group" "efs" {
+  name        = "${local.name}-efs"
+  description = "Allow inbound NFS traffic from private subnets of the VPC"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
```

Comment:
I believe we also should add UDP on 2049 as well

Reply:
Doc is pointing to only TCP. https://docs.aws.amazon.com/efs/latest/ug/accessing-fs-create-security-groups.html
Could you point me to more details here on why UDP is required?

Reply:
Oh, guess I am showing how dated I am 😅 - `While earlier versions of the protocol used User Datagram Protocol (UDP), from version 4.2 onwards NFS runs on top of Transmission Control Protocol (TCP).`

You used to need UDP https://aws.amazon.com/blogs/aws/amazon-efs-update-on-premises-access-via-direct-connect-vpc/ but I guess not anymore, please disregard!